libcpio/src/cpio.c

```diff
@@ -208,7 +208,7 @@ const void *cpio_get_file(const void *archive, unsigned long len, const char *na
         if (error) {
             return NULL;
         }
-        if (cpio_strncmp(header_info.filename, name, -1) == 0) {
+        if (cpio_strncmp(header_info.filename, name, ~0ul) == 0) {
```

Comment:
I wonder if an explicit cast might be a bit mote intuitive:

```suggestion
        if (cpio_strncmp(header_info.filename, name, (unsigned long)(-1)) == 0) {
```
Otherwise, if we have `ULONG_MAX` here:
```suggestion
        if (cpio_strncmp(header_info.filename, name, ULONG_MAX) == 0) {
```

Reply:
I was worried about some toolchain missing the "limits.h", but the repository already uses it, so `ULONG_MAX` it is.

Reply:
(according to [godbolt](godbolt.org) there is something called "ppci 0.5.5 compiler on CPython", which does not have it)

Reply:
Yea, we can't use limits.h because this library is used in a bootloader that doesn't provide a limits.h.

Reply:
You can see this in the failing compilation test: https://github.com/seL4/util_libs/actions/runs/4183101374/jobs/7247936534

I think explicitly casting to an unsigned long is more appropriate in this case.

Reply:
Updated. Though personally, I consider `~0ul` as concise as possible.